File: test_harness/test_suites/typescript_order_processing_v001/service.ts
import { Order, OrderImpl, OrderItemImpl, Product } from './models';
import { IOrderRepository } from './repository';

export class OrderService {
  constructor(private readonly repository: IOrderRepository) {}

  createOrder(customerId: string): Order {
    const order = new OrderImpl(customerId);
    this.repository.save(order);
    return order;
  }

  addItem(orderId: string, product: Product, quantity: number): Order {
    const order = this.repository.findById(orderId);
    if (!order) throw new Error(`Order ${orderId} not found`);
    const item = new OrderItemImpl(product, quantity);
    (order as OrderImpl).addItem(item);
    this.repository.save(order);
    return order;
  }

  confirmOrder(orderId: string): Order {
    const order = this.repository.findById(orderId);
    if (!order) throw new Error(`Order ${orderId} not found`);
    (order as OrderImpl).confirm();
    this.repository.save(order);
    return order;
  }

  getCustomerOrders(customerId: string): Order[] {
    return this.repository.findByCustomer(customerId);
  }
}

File: test_harness/test_suites/python_order_processing_v001/service.py
from typing import List, Optional
from models import Order, OrderItem, Product
from repository import IOrderRepository


class OrderService:
    def __init__(self, repository: IOrderRepository) -> None:
        self._repository = repository

    def create_order(self, customer_id: str) -> Order:
        order = Order(customer_id=customer_id)
        self._repository.save(order)
        return order

    def add_item(self, order_id: str, product: Product, quantity: int) -> Order:
        order = self._repository.find_by_id(order_id)
        if order is None:
            raise ValueError(f"Order {order_id} not found")
        if not product.is_available(quantity):
            raise ValueError(f"Insufficient stock for {product.name}")
        order.add_item(OrderItem(product=product, quantity=quantity))
        self._repository.save(order)
        return order

    def confirm_order(self, order_id: str) -> Order:
        order = self._repository.find_by_id(order_id)
        if order is None:
            raise ValueError(f"Order {order_id} not found")
        order.confirm()
        self._repository.save(order)
        return order

    def get_customer_orders(self, customer_id: str) -> List[Order]:
        return self._repository.find_by_customer(customer_id)
